src/main/resources/db/migration/0300/V321__AdHocPlotObservations.sql

```diff
@@ -0,0 +1,28 @@
+CREATE TABLE tracking.observation_types (
+    id INTEGER PRIMARY KEY,
+    name TEXT NOT NULL UNIQUE
+);
+
+INSERT INTO tracking.observation_types (id, name)
+VALUES (1, 'Monitoring'),
+       (2, 'Biomass Measurements')
+ON CONFLICT (id) DO UPDATE SET name = excluded.name;
+
+ALTER TABLE tracking.observations ADD COLUMN is_ad_hoc BOOLEAN;
+
+UPDATE tracking.observations
+SET is_ad_hoc = false
+WHERE is_ad_hoc IS NULL;
+
+ALTER TABLE tracking.observations ALTER COLUMN is_ad_hoc SET NOT NULL;
+
+ALTER TABLE tracking.observations ADD COLUMN observation_type_id INTEGER REFERENCES tracking.observation_types;
+
+UPDATE tracking.observations
+SET observation_type_id = 1
+WHERE observation_type_id IS NULL;
+
+ALTER TABLE tracking.observations ALTER COLUMN observation_type_id SET NOT NULL;
+
+ALTER TABLE tracking.monitoring_plots ALTER COLUMN planting_subzone_id DROP NOT NULL;
+ALTER TABLE tracking.observation_plots ALTER COLUMN monitoring_plot_history_id DROP NOT NULL;
```

Comment:
Dropping the not-null from monitoring plot subzone IDs is definitely needed both for this and for the flexible site editing changes.

I'm nervous about making plot history IDs optional. I think it's likely to force us to write more complicated queries for things like viewing historical ad-hoc observations on a site map since we can no longer count on observation plots being tied to particular map versions via history IDs (meaning we could get the correct map geometry with simple joins on the history IDs) and will instead need to calculate the right map to show by, I guess, comparing the observation times against ranges of modification times on the site history.

Also, I think it's plausible we'll allow users to rename or otherwise edit ad-hoc plots after the fact since we're letting users name them initially, which means they would have more than one history row.

Was this just about avoiding storing data that didn't seem necessary, or will it cause problems to keep history for these plots?